frontend/src/router/index.ts

```diff
@@ -140,15 +140,22 @@ export default function getRouter() {
       });
     }
 
+
+    // Uploading navigation guard
+    if (appStore.getIsUploading) {
+      if (!confirm('Navigation may cancel upload(s) in progress. ' +
+        'Please confirm you want to navigate from current page.')) {
+        return false;
+      }
+    }
+
     // Authentication Guard
     if (to.meta.requiresAuth) {
       const user = await authService.getUser();
       if (!user || user.expired) {
         router.replace({ name: RouteNames.LOGIN });
       }
     }
-
-    next();
```

Comment:
We shouldn't be changing the signature of the beforeEach lifecycle to be ignoring the next(). This callback should always be there to ensure that any subsequent lifecycle events are being invoked. By removing this, later lifecycle declarations many not ever be executed.